whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/domain/DomainIntersectionValidator.java

```diff
@@ -41,10 +45,11 @@ public DomainIntersectionValidator(final Ipv4DomainTree ipv4DomainTree,
     public void validate(final PreparedUpdate update, final UpdateContext updateContext) {
         final Domain domain = Domain.parse(update.getUpdatedObject().getKey());
         if (domain.getType() != INADDR) {
+            validateNserverCorrectPrefixes(update.getUpdate(), false);
             return;
         }
-
         validateIntersections(update, updateContext, (Ipv4Resource)domain.getReverseIp());
+        validateNserverCorrectPrefixes(update.getUpdate(), true);
     }
```

Comment:
I think prefix length check is for both ipv6 and ipv4. This handles only ipv4?

Reply:
not apply because now I am using the type of the Domain to know if it is ipv4 and ipv6